sys/include/net/netopt.h

```diff
@@ -868,6 +870,42 @@ typedef enum {
     /* add other states if needed */
 } netopt_state_t;
 
+/**
+ * @brief   Prepare context value for NETOPT_PDU_SIZE
+ *
+ * @param[in] src_len   length of the l2 src address
```

Comment:
Do we really need the `src_len`? Like the PAN IDs (which this function does not handle at all), they should be known by the device.

Reply:
How will the device know if a short or a long address will be used?

Reply:
the short address comes from a PAN coordinator. I think short source addresses can only be used when a device is associated to a PAN coordinator.

Reply:
Seeking for `macShortAddress` in the IEEE spec. I found the topic  6.7.1 Transmission:

![Screenshot_20230116_124655](https://user-images.githubusercontent.com/15147337/212671336-4bd2dab6-9ab2-44b9-84cb-333799bf74b3.png)

Reply:
But shouldn't the netedev layer not at least give you the option to use short addresses? It doesn't know what will be used on top.

Reply:
Currently I am looking for some more information. But so far, I think the decision which L2 source to choose depends on whether a device already has a short address as a result of association.
It looks like we build the short address from the two last bytes of the EUI-64.
`eui_short_from_eui64()`

Is there some reference for this? I have not looked for it so far.

Reply:
> Currently I am looking for some more information. But so far, I think the decision which L2 source to choose depends on whether a device already has a short address as a result of association.

This is correct. Associated devices on the same PAN use short source address. Beacons use extended short addresses.

Reply:
> It looks like we build the short address from the two last bytes of the EUI-64.
eui_short_from_eui64()

This is "the RIOT way", but in a real IEEE 802.15.4 network the coordinator would assign the short address. However, the application can still choose between using extended or short address (although the standard strongly suggests to use short address when possible).

At this point I think it would make more sense to add a "use short src address" flag rather than passing the actual source address length.

Reply:
The thing is that "use short src address" is specific to IEEE 802.15.4.
I could imagine that other MACs might implement this in the future and `gnrc_sixlowpan.c` wouldn't know about what's underneath it (at this point), but it knows about the L2 address length as that's set in `gnrc_netif_hdr_t`.

Reply:
hmmm but do we actually need to pass this information from the 6lowpan layer? why do we provide that information to the network interface if the MAC layer controls that anyway?
Shouldn't that be a configuration _below_ the interface?

Reply:
The 6lowpan layer needs to know how large the PDU for the given packet can be in order to fragment the IPv6 packet.

That's why it requests the PDU size when sending, for that it needs to provide the MAC layer with the transmission parameters.